src/Planner/PlannerJoinTree.cpp

```diff
@@ -182,6 +182,9 @@ bool applyTrivialCountIfPossible(
         return false;
 
     const auto & storage = table_node.getStorage();
+    if (!storage->supportsTrivialCountOptimization())
```

Comment:
This also fixes the `test_materialized_mysql_database` test under analyzer - https://s3.amazonaws.com/clickhouse-test-reports/52096/9d728161f1d8e82bdb7537faa53ca084fbc55643/integration_tests__asan__analyzer__[4_6].html